### PR TITLE
remove replaceable class from magic constant

### PR DESCRIPTION
**Pull request by amritpalnagrame**
#707 Fix as suggested by @kamil-tekiela.

### CODE REVIEW

**Review by sy-records — APPROVED** [review]

**Comment by kamil-tekiela** [comment]
I see now that the value was wrapped in `<replaceable>` and I wonder if that tag has any semantic meaning. In the manual it is replaced with `<span>` with a class that has no styles attached to it. I don't know about other formats, or if we should even worry about it. 

Does anyone know if it is safe to get rid of `<replaceable>` here?

File: language/constants.xml [hunk 1]
```diff
@@ -328,7 +328,7 @@ echo ANIMALS[1]; // outputs "cat"
         </entry>
        </row>
        <row xml:id="constant.coloncolonclass">
-        <entry><constant><replaceable>ClassName</replaceable>::class</constant></entry>
+        <entry><constant>::class</constant></entry>
```
Comment (cmb69) [inline language/constants.xml:331]: [`<replaceable>`](https://tdg.docbook.org/tdg/4.5/replaceable.html) conveys a meaning, and I think it's better to keep that as is (and maybe to improve PhD to somehow mark it up; perhaps as italics).

File: language/oop5/basic.xml [hunk 2]
```diff
@@ -621,7 +621,7 @@ Does\Not\Exist
     </example>
    </note>
    <para>
-    As of PHP 8.0.0, the <literal>::class</literal> constant may also be used on
+    As of PHP 8.0.0, the <literal>::class</literal> keyword may also be used on
```
Comment (cmb69) [inline language/oop5/basic.xml:624]: Maybe:
```suggestion
    As of PHP 8.0.0, the <constant><replaceable>ClassName</replaceable>::class</constant> magic constant may also be used on
```

Comment (kamil-tekiela) [reply language/oop5/basic.xml:624]: I'm on the fence. I'd rather use "keyword", which would be aligned with the rest of the page.

Comment (cmb69) [reply language/oop5/basic.xml:624]: Sorry, I lost context here, but still, calling `::class` a keyword isn't correct. Maybe either
```suggestion
    As of PHP 8.0.0, <literal>::class</literal> may also be used on
```
or
```suggestion
    As of PHP 8.0.0, <constant><replaceable>$object</replaceable>::class</constant> may also be used
```
and drop the "objects" on the next line.

Comment (kamil-tekiela) [reply language/oop5/basic.xml:624]: This is better.... but the `::class` can be used on other stuff too. See https://3v4l.org/6o6YA
Claiming that it only works on classes and objects, would be an understatement. It is a compile-time keyword/constant. Magic constants are generally run-time constants. `::class` is a category on its own with only one member.

Comment (cmb69) [reply language/oop5/basic.xml:624]: Hmm, right, although [most of this](https://3v4l.org/WP7Eb) is not particularly related to PHP 8. The first suggestion of my previous comment might be good for this paragraph. The case where the class does not exist is already catered to be the previous paragraph. Not sure what to do about string literals on the left hand side.

**Comment by afilina** [comment]
I addressed the @cmb69's comments, but that caused the `check-whitespace` job to fail on files unrelated to this PR. I attempted to rebase and force-push, which seems to have messed up the commit history instead. I reopened a new PR from the original branch here: https://github.com/php/doc-en/pull/2112

**Comment by Crell** [comment]
Fixed via #2112

**Pull request closed without merging**